prov/verbs/src/fi_verbs.h

```diff
@@ -394,33 +443,177 @@ void fi_ibv_monitor_unsubscribe(struct ofi_mem_monitor *notifier,
 				struct ofi_subscription *subscription);
 struct ofi_subscription *fi_ibv_monitor_get_event(struct ofi_mem_monitor *notifier);
 
+/*
+ * An XRC SRQ cannot be created until the associated RX CQ is known,
+ * maintain a list of validated pre-posted receives to post once
+ * the SRQ is created.
+ */
+struct fi_ibv_xrc_srx_prepost {
+	struct slist_entry	prepost_entry;
+	void			*buf;
+	void			*desc;
+	void			*context;
+	size_t			len;
+	fi_addr_t		src_addr;
+};
+
 struct fi_ibv_srq_ep {
 	struct fid_ep		ep_fid;
 	struct ibv_srq		*srq;
+	struct fi_ibv_domain	*domain;
+
+	/* Due to XRC SRQ semantics, an XRC SRX context may only be bound
+	 * to multiple endpoints when they share the same RX CQ. */
+	struct fi_ibv_cq	*srq_cq;
+
+	/* XRC SRQ is not created until endpoint enable */
+	fastlock_t		prepost_lock;
+	ofi_fastlock_acquire_t	pp_fastlock_acquire;
+	ofi_fastlock_release_t	pp_fastlock_release;
+
+	struct slist		prepost_list;
+	uint32_t		max_recv_wr;
+	uint32_t		max_sge;
+	uint32_t		prepost_count;
 };
```

Comment:
This is such a significant change to the structure, I would create a new xrc srq structure instead.

Reply:
@shefty - Since only a single SRQ is created by each RxM RDM_EP, are you fine with just embedding the XRC values within a XRC struct in the existing fi_ibv_srq_ep structure?

Reply:
I have shrunk this down to be just the pre-post list related elements and placed them within an embedded struct { } xrc; Is this sufficient?

Reply:
I'm good with this.  Apps shouldn't be allocating too many SRQs anyway.